File: core-mvp/src/main/java/ru/surfstudio/android/core/mvp/configurator/BindableScreenComponent.java
package ru.surfstudio.android.core.mvp.configurator;

import ru.surfstudio.android.core.mvp.presenter.Presenter;
import ru.surfstudio.android.core.mvp.view.PresenterHolderCoreView;

/**
 * Компонент для экранов с биндингом
 */
public interface BindableScreenComponent<V extends PresenterHolderCoreView> extends ScreenComponent<V> {

    Presenter[] initPresenter();
}

File: mvp/lib-core-mvp/src/main/java/ru/surfstudio/android/core/mvp/configurator/BindableScreenComponent.kt
package ru.surfstudio.android.core.mvp.configurator

import ru.surfstudio.android.core.mvp.view.PresenterHolderCoreView

/**
 * Компонент для экранов с биндингом.
 * Вызов [requestInjection] проинциализирует все зависимости c типом [Any] в даггер модуле.
 * Пример: используется для инициализации презентера, без явного указания @Inject на поле.
 */
interface BindableScreenComponent<V : PresenterHolderCoreView> : ScreenComponent<V> {

    fun requestInjection(): Any
}
